src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +192,20 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        # validate format of value for date variable
+        if action_type and action_type == LogicActionTypes.variable:
+            # get related form variable from context of parent serializer
+            form_var = self.parent.parent.context["form_variables"].variables[variable]
+            action_value = data.get("action", {}).get("value")
+
+            if form_var.data_type == FormVariableDataTypes.date:
+                if _ := format_date_value(action_value) == "":
+                    raise serializers.ValidationError(
+                        {
+                            "action.value": "Value for date variable must be in the format yyyy-mm-dd"
```

Comment:
this error message structure should be a nested structure, the formatting with dotted paths is something that is done in the generic validation error handler -> I expected a nested dict here instead.